app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/FriendsActivityTest.java

```diff
@@ -0,0 +1,218 @@
+        return new ViewAction() {
+            @Override
+            public Matcher<View> getConstraints() {
+                return constraints;
+            }
+
+            @Override
+            public String getDescription() {
+                return action.getDescription();
+            }
+
+            @Override
+            public void perform(UiController uiController, View view) {
+                action.perform(uiController, view);
+            }
+        };
+    }
+
+    /* Set up the environment */
+    @BeforeClass
+    public static void init() throws InterruptedException {
+        /* Make sure that mock users are not on the database before the tests*/
+        for(int i=0; i < FRIENDS_SIZE; i++) {
+            Database.refRoot.child("users").child("test" + i).removeValue();
+        }
+        Database.refRoot.child("users").child("null").removeValue();
+        Thread.sleep(1500);
+
+        /* Make sure no user is signed in before a test */
+        FirebaseAuth.getInstance().signOut();
+        Account.getAccount().synchronizeUserProfile();
+        Thread.sleep(1500);
+    }
+
+    /* Make sure that mock users are not on the database after a test */
+    @After
+    public void removeTestUsers() throws InterruptedException {
+        for(int i=0; i < FRIENDS_SIZE; i++) {
+            Database.refRoot.child("users").child("test" + i).removeValue();
+        }
+        Database.refRoot.child("users").child("null").removeValue();
+        Thread.sleep(1500);
+    }
+
+    @Rule
+    public ActivityScenarioRule<FriendsActivity> testRule = new ActivityScenarioRule<>(FriendsActivity.class);
+
+    /* Create Intent */
+    @Before
+    public void setup(){
+        Intents.init();
+    }
+
+    /* Release Intent */
+    @After
+    public void cleanUp(){
+        Intents.release();
+    }
+
+    /* Test that the toolbar title is set as expected */
+    @Test
+    public void TestToolbarTitle(){
+        String TOOLBAR_TITLE = "Friends";
+        ViewInteraction greetingText = Espresso.onView(withId(R.id.toolbarTitle));
+        greetingText.check(matches(withText(TOOLBAR_TITLE)));
+    }
+
+    /* Test that the activity finishes when the toolbar back button is pressed. */
+    @Test
+    public void TestToolbarBackButton(){
+        onView(withId(R.id.toolbarBackButton)).perform(click());
+        assertSame(testRule.getScenario().getResult().getResultCode(), Activity.RESULT_CANCELED);
+    }
+
+    /* Test that all elements in list view are at correct place and contains correct data */
+    @Test
+    public void TestContentOfListView() throws InterruptedException {
+        // Set up the scenario
+        // Add mock users
+        String TESTING_USERNAME = "username@Test";
+        for(int i=0; i < FRIENDS_SIZE; i++) {
+            Database.setChild(Database.CHILD_USERS + "test" + i, Collections.singletonList(Database.CHILD_USERNAME), Collections.singletonList(TESTING_USERNAME + i));

```

Comment:
Maybe more of a TODO comment to be added, a bit out of scope. But should we add a class to handle this so we can call a "addFriend" function instead of this? What do you think?